### PR TITLE
PFE-7 Update component links

### PR DESCRIPTION
**Pull request by Joelhagg**
Update the link component for payex design guide as new ux figma design

Case PFE-7 in jira

## Description

This PR will include an update to the Link component and its colors. The issue and its fix was that colors when hovering on the "bright color" mode on the link turned green.

## How Has This Been Tested?

Test passed:
image

## Types of changes

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [ ] I have updated the **CHANGELOG** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

## Review instructions

[Review instructions](../REVIEW_INSTRUCTIONS.md)

### CODE REVIEW

**Comment by goldenraphti** [comment]
@Joelhagg Thanks a lot for your contribution !!! 🌟 🚀 
It seems great 👍 

1 question & 1 suggestion:
- there is no visual feedback anymore when user hovers the link on non-bright color alternative (beside cursor switching to a pointer). Is that the intended result?
- suggestion: could you replace the "LESS variables" you've introduced with CSS custom properties? The idea is to slowly migrate to it, and hopefully one day leave preprocessors behind 😅 . Variables for PayEx are there: "src/less/variables-payex.less" They should pretty much all exist in both LESS and pure CSS variables I think 😉 I'm not saying to also change the ones you haven't touched, just the ones you're introducing 😉 For the future tickets you'll be puching (which I hope they'll be plenty 🤩 ) it's only a rule if it's not meaning creating new CSS custom properties & not too complicated (e.g. in some places some LESS variables are fed to LESS functions to iterate over a list of colors. Then nevermind, use as is and I'll migrate them myself in a dedicated effort )

Thanks again for your time ! 💫

File: src/less/components/payex/links.less [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
 		color: @link-hover-color;
+		text-decoration: none;
 	}
 
 	&:focus {
 		color: @brand-primary;
 	}
 
+	&.bright {
+		color: @payex-light-blue;
```
Comment (goldenraphti) [inline src/less/components/payex/links.less:15]:
```suggestion
		color: var(--payex-light-blue);
```

File: src/less/components/payex/links.less [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+
+		i {
+			color: @payex-light-blue;
```
Comment (goldenraphti) [inline src/less/components/payex/links.less:18]:
```suggestion
			color: var(--payex-light-blue);
```

File: src/less/components/payex/links.less [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+		}
+
+		&:focus-visible {
+			outline: 2px solid @white;
```
Comment (goldenraphti) [inline src/less/components/payex/links.less:22]:
```suggestion
			outline: 2px solid var(--white);
```

File: src/less/components/payex/links.less [hunk 4]
```diff
@@ -1,28 +1,46 @@
...
+
 	&.icon-link {
-		&:hover {
-			color: @brand-primary;
+
+		&.bright {
 
 			span {
-				text-decoration: none;
+				color: @payex-light-blue;
```
Comment (goldenraphti) [inline src/less/components/payex/links.less:32]:
```suggestion
				color: var(--payex-light-blue);
```

File: src/less/components/payex/links.less [hunk 5]
```diff
@@ -1,28 +1,46 @@
...
+
+				&:hover {
+					text-decoration: underline;
+				}
 			}
 		}
 
-		:focus {
+		&:hover {
+			color: @brand-primary;
```
Comment (goldenraphti) [inline src/less/components/payex/links.less:41]:
```suggestion
			color: var(--brand-primary);
```

**Comment by Joelhagg** [comment]
> 1 question & 1 suggestion:
> 
> * there is no visual feedback anymore when user hovers the link on non-bright color alternative (beside cursor switching to a pointer). Is that the intended result?

_I made som changes after the UX Design in Figma was updated and you were right about the hover state!_

> * suggestion: could you replace the "LESS variables" you've introduced with CSS custom properties? The idea is to slowly migrate to it, and hopefully one day leave preprocessors behind 😅 . Variables for PayEx are there: "src/less/variables-payex.less" They should pretty much all exist in both LESS and pure CSS variables I think 😉 I'm not saying to also change the ones you haven't touched, just the ones you're introducing 😉 For the future tickets you'll be puching (which I hope they'll be plenty 🤩 ) it's only a rule if it's not meaning creating new CSS custom properties & not too complicated (e.g. in some places some LESS variables are fed to LESS functions to iterate over a list of colors. Then nevermind, use as is and I'll migrate them myself in a dedicated effort )

_I accepted your review changes and made some new updates so that it uses CSS Custom property's_
_Also added 2 new color variables_

**Comment by Joelhagg** [comment]
> @Joelhagg Looks good, might add padding in focus state to the sides: image

Will do in future case 👌

**Review by WMeander — APPROVED** [review]

**Review by goldenraphti — APPROVED** [review]

**Comment by goldenraphti** [comment]
I'll merge it once [SWED-2276 ](https://github.com/SwedbankPay/design.swedbankpay.com/pull/927) gets approved and merged (hopefully today 😉) 👍

**Pull request merged**